internal/registry/source.go

```diff
@@ -81,17 +84,64 @@ func GetPresetRegistryInfo(kind string) string {
 }
 
 var (
-	Home     = "home"
-	Author   = "_author"
-	Registry = "registry"
-	Delete   = "delete"
-	Default  = "DEFAULT"
-	Nrmrc    = path.Join(GetSystemPreffix(), ".nrmrc")
-	Npmrc    = path.Join(GetSystemPreffix(), ".npmrc")
+	Grmrc = path.Join(GetSystemPreffix(), ".grmrc.yaml")
+	Npmrc = path.Join(GetSystemPreffix(), ".npmrc")
 )
 
 type RegistryDataSource struct {
-	Registry     map[string]string
-	Keys         []string
-	UserRegistry map[string]string
+	fs         fs.FS
+	Registry   map[string]string
+	Keys       []string
+	PresetKeys []string
+	Niave      map[string]RegsitryInfo
+}
+
+func (r *RegistryDataSource) Drop(name string) error {
+	if _, ok := r.Niave[name]; ok {
+		delete(r.Niave, name)
+		parsed := parsr(r.Niave)
+		err, _ := r.fs.WriteYAML(Grmrc, parsed)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+	return errors.New("Not found")
```

Comment:
"not found" is a common error, we could define a const error in package.

e.g.
```
const NOTFOUND_ERROR = errors.New("Not found")
```

 In this way, we can return xxx.NOTFOUND_ERROR

Reply:
Right, I will use static enum replace it.